sdk/resourcemanager/Azure.ResourceManager/src/Custom/Resources/Models/Location.cs

```diff
@@ -7,12 +7,14 @@
 using System.Globalization;
 using System.Text;
 using System.Text.RegularExpressions;
+using Azure.ResourceManager.Core;
 
 namespace Azure.ResourceManager.Resources.Models
 {
     /// <summary>
     /// Represents an Azure geography region where supported resource providers live.
     /// </summary>
+    [PropertyReferenceType]
     public partial class Location : IEquatable<Location>, IComparable<Location>
```

Comment:
If we are planning to replace the type with this "Location" in generated code, if all the properties match(similar to [ResourceIdentity](https://github.com/Azure/azure-sdk-for-net/blob/96ec8bf2a432bf58d7dd8d776e07d7dac7c892f5/sdk/resourcemanager/Azure.ResourceManager/src/Custom/Resources/Models/ResourceIdentity.cs#L15-L16)), then in order to access serialization and deserialization methods, we have to add json converter. Because our [serialization and deserialization methods are internal](https://github.com/Azure/azure-sdk-for-net/blob/96ec8bf2a432bf58d7dd8d776e07d7dac7c892f5/sdk/resourcemanager/Azure.ResourceManager/src/Generated/Resources/Models/Location.Serialization.cs#L15).
Check below for an example:
https://github.com/Azure/azure-sdk-for-net/blob/96ec8bf2a432bf58d7dd8d776e07d7dac7c892f5/sdk/resourcemanager/Azure.ResourceManager/src/Custom/Resources/Models/ResourceIdentity.Serialization.cs#L12.